tasks/main.yml

```diff
@@ -66,3 +66,35 @@
   changed_when: sonar_version | string not in api_check_version.content
   notify: fail if expected sonar version not confirmed
   ignore_errors: "{{ ansible_check_mode }}"
+
+- name: "Ensure old admin password match"
+  uri:
+    method: POST
+    url: "http://{{ sonar_web_host }}:{{ sonar_web_port }}{{ sonar_web_context }}/api/authentication/login"
+    force_basic_auth: true
+    url_username: "{{ sonar_admin_user }}"
+    url_password: "{{ sonar_admin_old_password }}"
+    body: "login={{ sonar_admin_user }}&password={{ sonar_admin_old_password }}"
+    body_format: form-urlencoded
+    validate_certs: false
+    status_code:
+      - 200
+      - 401
+  register: login_check
+  when:
+    - sonar_admin_password_change
+
+- name: "Change old admin password new one"
+  uri:
+    method: POST
+    url: "http://{{ sonar_web_host }}:{{ sonar_web_port }}{{ sonar_web_context }}/api/users/change_password"
```

Comment:
Is it safe to assume the protocol will always be http ?

Reply:
Maybe you right, but you should run nginx before sonarqube installation, so queries to http:127.0.0.1:9000 will be secured